src/main/java/com/google/sps/servlets/FindLocationServlet.java

```diff
@@ -19,7 +22,15 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     
-    //TODO - implement the rest of this function (@Kaleab)
+    final String correlator = DBUtilities.getCorrelator(request);
+
+    if (correlator.isEmpty()) {
+      response.setStatus(HttpServletResponse.SC_NOT_FOUND);
```

Comment:
This would execute if they don't pass in a correlator right?
But you're returning a 404, which is usally used for when an entity isn't found, meaning like that url doesn't exist.  This should probably be a 400 - Bad Request.

A 404 could be used if they send in a correlator, but it can't be found in the db anywhere